### PR TITLE
Implement Generic Objects on services details page

### PR DESCRIPTION
**Pull request by chalettu**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1497705

### CODE REVIEW

**Comment by chalettu** [comment]
@miq-bot add_label enhancement
@miq-bot add_label fine/no

**Comment by chalettu** [comment]
Here is a sample screenshot of the implementation of Generic Objects, @chriskacerguis .  Keep in mind this does not currently show the custom actions drop down yet.  
demo_go

**Comment by chriskacerguis** [comment]
@chalettu can you please attach the issue to this PR?

**Comment by chalettu** [comment]
https://github.com/ManageIQ/manageiq-ui-service/issues/934

**Comment by bascar** [comment]
@chalettu see https://github.com/ManageIQ/manageiq/pull/16006 as relates to the picture question above

**Comment by serenamarie125** [comment]
@chalettu looking good.  I know this is WIP, but since it has the ux/review tag i assume you want me to provide feedback on what you have

1. The main accordion showing the generic object type should have parens with the # of generic objects of that type, similar to the Compute resources shown above.
2. the collapsed Generic Object List view item should show the Generic object type icon before the name of the object
3. is it possible for you to put 4 key/value pairs for mock data so that we can see the alignment of the information ?

**Comment by Loicavenel** [comment]
@bascar do you have some example of Key/Value = Generic Object Properties we may have.. Today we have 3 on one line and Serena is suggesting may be 4..  Do you know what is the limit engineering put in the DB for a a key and a Value.. 255?

**Comment by serenamarie125** [comment]
@chalettu @Loicavenel @bascar regarding my #3 comment, I didn't mean we should show 4 pairs on a row, having 3 on a row is sufficient.

I'm suggesting that we have dummy data for 4 key/value pairs that are used and added as a mockup to understand what it looks like.  It would be great to have some real examples from Brad so that Chris can show them here though!

**Comment by bascar** [comment]
There is no limit to the pairs as it is all a JSON payload made to look like attributes. 
Examples of some could be:
arn => arn:aws:rds:eu-west-1:123456789012:db:mysql-db
dept => Finance
cost_center => 4567890
retry_attempts => 123
cleanup_playbook=> https://github.com/somproject/helpers/blob/master/tools/somecleanupscript.yml
......

**Comment by chalettu** [comment]
Currently awaiting https://github.com/ManageIQ/manageiq-api/pull/57 that will allow subcollections on Generic Objects
Also currently awaiting the Generic Object custom actions functionality to be completed.  No PR as of yet.

**Comment by chriskacerguis** [comment]
@chalettu this needs to get bumped in test coverage :)

**Comment by chriskacerguis** [comment]
@bascar @Loicavenel FYI - This is blocked until ManageIQ/manageiq-api#57 is merged

**Comment by bascar** [comment]
I asked nicely for a merge :)

Best Regards,
BRAD ASCAR

CLOUDFORMS PRODUCT MANAGER - PUBLIC CLOUD

Red Hat  <https://www.redhat.com/>

bascar@redhat.com    M: +1-770-371-3719
<http://redhatemailsignature-marketing.itos.redhat.com/>
<https://red.ht/sig>
TRIED. TESTED. TRUSTED. <https://redhat.com/trusted>

**Comment by bascar** [comment]
I asked nicely for a merge, thanks @chriskacerguis

**Comment by chriskacerguis** [comment]
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1497705

**Comment by chriskacerguis** [comment]
@chalettu could we add some tests for this?

**Comment by chalettu** [comment]
@chriskacerguis we are also dependent on https://github.com/ManageIQ/manageiq-api/pull/93 for implementing custom actions on Generic Objects

**Comment by serenamarie125** [comment]
Is there any way we can see what this will look like with more pairs?  Brad provided some.  It would be nice to see what it'll look like.  I'm concerned about alignment, and would rather see that now if possible than later.

**Comment by chalettu** [comment]
updated go preview
@serenamarie125 here is the latest Screenshot of GO that implements custom actions. Please let me know if we have any revisions we would like to make.

**Comment by chalettu** [comment]
We are going to add generic object icon support after https://github.com/ManageIQ/manageiq-api/pull/114 gets merged in.   This will allow us to show custom pictures

**Comment by serenamarie125** [comment]
@chalettu the reason I asked about seeing what this looks like with more pairs of data is because my mocks (shown [here](https://docs.google.com/document/d/1oBVIJNk-X_XoUDun2RHpg2Hp0Sp6ljwPrGxGlV0UyfE/) have the label left aligned rather than right aligned as you have implemented.  This may be perfectly acceptable, but I wanted to see what it looks like.

I also don't see the generic object type icon, can you access that?

**Comment by chalettu** [comment]
generic_objects_final
@serenamarie125 here is a updated screenshot that includes Generic object picture support and also for multiple generic object attributes.  Please let me know if you approve of the UI.

**Comment by chalettu** [comment]
@miq-bot remove_label wip

**Comment by chriskacerguis** [comment]
@serenamarie125 I spoke to @chalettu this is good for your review :)

/cc @bascar @Loicavenel

**Comment by serenamarie125** [comment]
@miq-bot add_labels ux/approved

**Comment by serenamarie125** [comment]
@miq-bot remove_labels ux/review

File: client/app/services/generic-objects-list/generic-objects-list.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<pf-list-view
+					config="vm.genericObjectsListConfig"
+					items="vm.genericObject.objects"
+					custom-scope="vm">
+					<div class="row generic-objects-container">
+            <div class="col-lg-11 col-md-11 col-sm-11 col-xs-11 generic-objects-title"
+                 ng-click="$ctrl.customScope.toggleExpand(item)">
+              <i class="fa fa-angle-right"
+                 ng-class="{'fa-angle-down': item.isExpanded}"></i>
+                 <img src="{{item.picture.image_href}}" class="generic-object-icon" ng-if="item.picture">
```
Comment (AllenBW) [inline client/app/services/generic-objects-list/generic-objects-list.html]: prefer use of `ng-src` here, while this probably will work most of the time, this should be done for the one time the browser will still try to load the {{path}} string it finds in ``

**Pull request merged**